src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java

```diff
@@ -849,18 +849,25 @@ public final void markDirty()
      * register a block and position.
      *
      * @param block to be registered
-     * @param pos of the block
+     * @param pos   of the block
      */
-    public void registerBlockPosition(@NotNull Block block, @NotNull final BlockPos pos)
+    public void registerBlockPosition(@NotNull Block block, @NotNull final BlockPos pos, @NotNull final World world)
     {
-        if (block instanceof BlockContainer)
+
+        TileEntity tile = world.getTileEntity(pos);
+
+        if (tile != null)
         {
-            addContainerPosition(pos);
+            if (tile.hasCapability(CapabilityItemHandler.ITEM_HANDLER_CAPABILITY, null))
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1066)